### PR TITLE
Fix: support `doc_pk` storage path placeholder via API

### LINKED ISSUE #4178: [BUG] Invalid variable detected for "doc_pk"
### Description

Currently, it is not possible to use the new variable "doc_pk" in storage paths. An error is encountered "Invalid variable detected.". I would like to use this variable in my file naming options

image

### Steps to reproduce

1. Go to "Manage / Storage paths"
2. Click on to "Create"
3. Assign a name to the path
4. Assign a path with the variable "doc_pk" (eg. {doc_pk} or {created_year}/{created}-{doc_pk}

### Webserver logs

```bash
[WARNING] [django.request] Bad Request: /api/storage_paths/1/
```

### Browser logs

```bash
{"headers":{"normalizedNames":{},"lazyUpdate":null},"status":400,"statusText":"Bad Request","url":"http://coyotte:8002/api/storage_paths/","ok":false,"name":"HttpErrorResponse","message":"Http failure response for http://coyotte:8002/api/storage_paths/: 400 Bad Request","error":{"path":["Invalid variable detected."]}}
```

### Paperless-ngx version

1.7.4

### Host OS

Ubuntu 23.04

### Installation method

Docker - official image

### Browser

Firefox

### Configuration changes

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by shamoon**
## Proposed change

Fixes #4178

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please explain):

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [ ] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Review by stumpylog — APPROVED** [review]
Woops, missed this in the original PR for sure

File: src/documents/tests/test_api.py [hunk 1]
```diff
@@ -4463,7 +4463,7 @@ def test_api_storage_path_placeholders(self):
                     "/{created_month_name_short}/{created_day}/{added}/{added_year}"
                     "/{added_year_short}/{added_month}/{added_month_name}"
                     "/{added_month_name_short}/{added_day}/{asn}/{tags}"
-                    "/{tag_list}/",
+                    "/{tag_list}/{owner_username}/{original_name}/{dock_pk}/",
```
Comment (hermannoffen) [inline src/documents/tests/test_api.py:4466]: Here likewise.

File: src/documents/serialisers.py [hunk 2]
```diff
@@ -835,6 +835,7 @@ def validate_path(self, path):
                 tag_list="tag_list",
                 owner_username="someone",
                 original_name="testfile",
+                dock_pk="doc_pk",
```
Comment (hermannoffen) [inline src/documents/serialisers.py:838]: Shouldn't it be `doc_pk="doc_pk",`?

**Review by hermannoffen — COMMENTED** [review]
There might be a typo.

File: src/documents/serialisers.py (hunk 2 above)
Comment (shamoon) [reply src/documents/serialisers.py:838]: Wow thanks great catch, I'll fix that

**Pull request merged**